docs/src/developer/future.md

```diff
@@ -0,0 +1,127 @@
+```@meta
+CurrentModule = Nemo
+```
+
+# Future plans
+
+## Ring and CommRing
+
+Currently all commutative ring types belong to `Ring` and their elements to
+`RingElem` (and `RingElement`) and we have separate types for noncommutative
+rings and elements thereof, i.e. `NCRing` and `NCRingElem` etc.
+
+However, it would be more logical to use `Ring` for not necessarily
+commutative rings and `CommRing`, `CRing` or `CommutativeRing` (the name has
+not been decided on yet) for commutative rings.
+
+This is a big change and should happen with plenty of warning for the
+community. It would be convenient if a script could be made available to
+automate this.
+
+## Matrices and polynomials over number fields
+
+The Antic C library is used for number field elements, but matrices and
+polynomials over these are handled by Julia. There is nothing explicitly
+wrong with this, except that the jit compilation is costly for such basic
+operations and the code is not as easy for others to use in other projects
+e.g. if they are using a different language or do not want additional
+dependencies outside the Antic C library they are using.
+
+We plan to write some routines in C in the Antic library to handle at least
+univariate polynomials over number fields and matrices as well. These should
+use highly optimised routines based on the algorithms with best complexity
+for these operations.
```

Comment:
I wonder if alternatives were considered and tested with some quick experiments, e.g. using `@nospecialize` ? Writing yet more C code is something that require resources and time, and we should be reaaally sure that it will have a noticeable benefit.

As to priority: do we have an immediate application where this is relevant, or is this more out of a desire to win in artificial benchmarks against other implementations? (The latter certainly is not unimportant, but IMHO less of a priority than the former, esp. given that we have very limited resources and need to prioritize)

Reply:
This will happen. I will be implementing it. It has nothing to do with speed but to do with what I need to exist in C.

Reply:
I don't mind if this is happening or not, but I am not sure this has to be included as future plans for Nemo/AbstractAlgebra. We are also not listing all the plans for flint or arb here.

Reply:
Ok, I can remove it. I just felt that it would be worthwhile to inform people of things that will be happening. I can post it on the Nemo devel list instead.

Reply:
But what is the relation of this is to Nemo? Isn't this specific to the Antic library?

Reply:
I've removed this section already. No need to discuss further.